components/motor/tmcstepper/stepper_motor_tmc.go

```diff
@@ -60,7 +66,6 @@ func (config *TMC5072Config) Validate(path string) ([]string, error) {
 	if config.TicksPerRotation <= 0 {
 		return nil, utils.NewConfigValidationFieldRequiredError(path, "ticks_per_rotation")
 	}
-	deps = append(deps, config.BoardName)
```

Comment:
seems like moving this to what is now line 55 is totally fine, but I'm wondering if it could cause any issues. is there a situation in which you want to know if the SPIBus exists before appending the board to the dependencies? I wouldn't think so but just want to make sure

Reply:
I don't think so. If the bus doesn't exist, the config to change is this one, and fiddling with the board component's config won't do anything. Similarly, if you fiddle with the board component's config, it won't affect anything over here unless we're using a GPIO pin.

I appreciate you double-checking my assumptions, though!